infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -497,7 +514,9 @@ public void updateSiteConfig(LogReplicationClusterInfo.TopologyConfigurationMsg
      */
     @Override
     public void prepareSiteRoleChange() {
-        replicationManager.prepareSiteRoleChange();
+        if (localClusterDescriptor.getRole() == ClusterRole.ACTIVE && replicationManager != null) {
```

Comment:
May add log.warn that this call should not be called on non active site. Also why replicationManager can be null?

Reply:
Added. If the replication manager is not initialized yet, it will be null. It happens when the cluster role is standby.